Sources/Player/Player/Player+ControlCenter.swift

```diff
@@ -127,9 +127,9 @@ extension Player {
     }
 
     func nowPlayingPublisher() -> AnyPublisher<NowPlaying, Never> {
-        Publishers.CombineLatest(isActivePublisher, queuePublisher)
-            .map { [weak self] isActive, queue in
-                guard let self, isActive, !queue.isActive else { return Just(NowPlaying.empty).eraseToAnyPublisher() }
+        Publishers.CombineLatest3(isActivePublisher, queuePublisher, queueErrorPublisher())
+            .map { [weak self] isActive, queue, error in
+                guard let self, isActive, !queue.isActive, error == nil else { return Just(NowPlaying.empty).eraseToAnyPublisher() }
```

Comment:
The queue publisher already reacts to errors, so we should simply test `queue.error` in the map operator closure here. I checked this intuition and the test you added passes.

```suggestion
                guard let self, isActive, !queue.isActive, queue.error == nil else { return Just(NowPlaying.empty).eraseToAnyPublisher() }
```

If this matches your expectation as well I guess we can then revert the other changes since they are not required.